File: weakreferrencedemo/src/main/java/com/example/weakreferrencedemo/MainActivity.java
package com.example.weakreferrencedemo;

import androidx.appcompat.app.AppCompatActivity;
import android.os.Bundle;

import java.lang.ref.WeakReference;

public class MainActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

    }
}

File: kotlindemo/src/main/java/com/example/kotlindemo/MainActivity.kt
package com.example.kotlindemo

import android.content.Intent
import android.os.Bundle
import android.support.v4.app.ActivityOptionsCompat
import android.support.v7.app.AppCompatActivity
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        tv_phone_login.setOnClickListener {
//            Toast.makeText(this, "login", Toast.LENGTH_SHORT).show()
            val intent = Intent(this,LoginActivity::class.java)
            val  options = ActivityOptionsCompat.makeSceneTransitionAnimation(this,tv_phone_login,"phonelogin")
            startActivity(intent,options.toBundle())
        }
    }
}
